trace/redis/redis.go

```diff
@@ -0,0 +1,67 @@
+package redis
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/beatlabs/patron/trace"
+	"github.com/go-redis/redis/v7"
+	"github.com/opentracing/opentracing-go"
+)
+
+type connInfo struct {
+	instance string
+}
+
+func (c *connInfo) startSpan(ctx context.Context, opName, stmt string) (opentracing.Span, context.Context) {
+	return trace.RedisSpan(ctx, opName, trace.RedisComponent, trace.RedisDBType, stmt, c.instance)
+}
+
+// Client wraps redis.Client for easier usage.
+type Client redis.Client
+
+// Options wraps redis.Options for easier usage.
+type Options redis.Options
+
+// Empty represents the error which is returned in case a key is not found.
+const Empty = redis.Nil
+
+// Conn represents a connection with a Redis client.
+type Conn struct {
```

Comment:
This should be the other way around.
A client owns a connection.
And the methods like `Do` should be on the client not the connection.

Reply:
Okay, got it. Basically, the thing that go-redis exposes and acts upon is the Client, so maybe the Connection could be dropped altogether.   
I initially had added it so that the API matched that of `trace/sql`, but I'll turn things around and see how it looks.